ompi/datatype/ompi_datatype_create_vector.c

```diff
@@ -49,7 +49,7 @@ int32_t ompi_datatype_create_vector( int count, int bLength, int stride,
 
     pData = ompi_datatype_create( oldType->super.desc.used + 2 );
     if( (bLength == stride) || (1 >= count) ) {  /* the elements are contiguous */
-        ompi_datatype_add( pData, oldType, count * bLength, 0, extent );
+        ompi_datatype_add( pData, oldType, (size_t)count * bLength, 0, extent );
```

Comment:
From C99 standard 6.3.1.8:
> Otherwise, if the operand that has unsigned integer type has rank greater or equal to the rank of the type of the other operand, then the operand with signed integer type is converted to the type of the operand with unsigned integer type.

My reading of this paragraph indicates that in this particular case the expression promotion goes from left to right, in which case there is no need to cast all members.

Reply:
thanks for the explanation, I will approve the PR.

feel free to merge it as-is, or to cast `bLength` as well. I understand it is not required, but that can be seen as defensive programming.